contracts/cw721-base/src/state.rs

```diff
@@ -3,42 +3,47 @@ use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 use std::marker::PhantomData;
 
-use cosmwasm_std::{Addr, BlockInfo, StdResult, Storage};
+use cosmwasm_std::{Addr, BlockInfo, CustomMsg, StdResult, Storage};
 
-use cw721::{ContractInfoResponse, CustomMsg, Cw721, Expiration};
+use cw721::{ContractInfoResponse, Cw721, Cw721Query, Expiration};
 use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
 
-pub struct Cw721Contract<'a, T, C, E, Q>
+pub struct Cw721Contract<'a, T, C, I, E, Q>
 where
     T: Serialize + DeserializeOwned + Clone,
+    I: CustomMsg + DeserializeOwned,
     Q: CustomMsg,
     E: CustomMsg,
 {
-    pub contract_info: Item<'a, ContractInfoResponse>,
+    pub contract_info: Item<'a, ContractInfoResponse<I>>,
     pub minter: Item<'a, Addr>,
     pub token_count: Item<'a, u64>,
     /// Stored as (granter, operator) giving operator full control over granter's account
     pub operators: Map<'a, (&'a Addr, &'a Addr), Expiration>,
     pub tokens: IndexedMap<'a, &'a str, TokenInfo<T>, TokenIndexes<'a, T>>,
 
     pub(crate) _custom_response: PhantomData<C>,
+    pub(crate) _custom_instantiate: PhantomData<I>,
     pub(crate) _custom_query: PhantomData<Q>,
     pub(crate) _custom_execute: PhantomData<E>,
 }
 
 // This is a signal, the implementations are in other files
-impl<'a, T, C, E, Q> Cw721<T, C> for Cw721Contract<'a, T, C, E, Q>
+impl<'a, T, C, I, E, Q> Cw721<T, C> for Cw721Contract<'a, T, C, I, E, Q>
 where
     T: Serialize + DeserializeOwned + Clone,
     C: CustomMsg,
+    I: CustomMsg + DeserializeOwned,
     E: CustomMsg,
     Q: CustomMsg,
+    Cw721Contract<'a, T, C, I, E, Q>: Cw721Query<T, C>,
```

Comment:
Something about this line irks me. Feel it shouldn't be needed... but doesn't work without it.

Reply:
Think this should be:

```suggestion
    Cw721Contract<'a, T, C, I, E, Q>: Cw721Query<T, I>,
```

In `Cw721Query` the second generic corresponds to `ContractInfoResponse<I>`. By putting `C` here we're saying this implements the query interface where the contract level metadata extension is the same as the extension used for `CosmosMsg` in responses.

Could be mixing things up here, I find this quite hard to assemble in my brain. :) More reason to give these generics actual names lol.

Reply:
Also, we can just remove this entire block. :) Tested locally that the build still works.

This makes sense as we provide implementations of both `Cw721Query` and `Cw721Execute` in `query.rs` and `execute.rs` respectively. Not even sure what telling the compiler this again accomplishes.

Reply:
Lol, that's great! Nice. : )